c/column_fw.cc

```diff
@@ -141,7 +141,7 @@ void FwColumn<T>::reify() {
   // If the rowindex is absent, then the column is already materialized.
   if (!ri) return;
   bool simple_slice = ri.isslice() && ri.slice_step() == 1;
-  bool ascending    = ri.isslice() && ri.slice_step() > 0;
+  bool ascending = ri.isslice() && static_cast<int64_t>(ri.slice_step()) > 0;
```

Comment:
When I look at return type of `ri.slice_step()`, it is `size_t`. Then what is your definition of a negative slice step?

Reply:
For example `step = 18446744073709551615 = size_t(-1)` is considered negative, because for any reasonable starting index `start`, we will have `start + step < start` using `size_t` arithmetics. I can test whether the step is negative by casting into `int64_t` and checking the sign of that.

Reply:
I see, so you're using `size_t` arithmetics everywhere for the slices, and basically `std::numeric_limits<size_t>::max() - i` will result in `-(i+1)` step?

Reply:
Yes, though I don't do subtraction in code, just casting between `int64_t` and `size_t`